### PR TITLE
Disabled the animation during page load

### PR DESCRIPTION
**Pull request by promatik**
I can think about two ways of fixing this;
- Adding an ajax request to tell the backend if the sidebar is closed or not (kind of heavy)
- Disabling the animation during that first sidebar close.

I went for the second one.

https://user-images.githubusercontent.com/1838187/105838832-8d151880-5fc8-11eb-8ea0-ce64cfdcd0f1.mp4

### CODE REVIEW

**Comment by pxpm** [comment]
Hello @promatik 

I am pretty sure the flick still happens, it's just much faster and without the animation it's unoticeable. 

But when you say `We migth use ajax request to the backend` is to: "open the sidebar in correct state" ? If that is the case, could we use `cookies` instead of local storage for preserving sidebar state ? 

Let me know,
Pedro

**Comment by promatik** [comment]
> I am pretty sure the flick still happens, it's just much faster and without the animation it's unnoticeable.

@pxpm, yes it's true, but, the big advantage here is that this javascript is Vanilla JS, and runs before DOM ready, (no onDomReady listener).
So it's very unlikable (if not impossible) the flick happens to anyone, because backpack depends on jQuery an other frameworks that need the dom to be ready before rendering the page.

If you think we should move to the cookie solution, I agree that would be the best option, but that requires an extra route, and an ajax request each time the user changes the sidebar state, to tell the PHP to store the cookie with that info.

Let me know what you think ✌

**Comment by pxpm** [comment]
Sorry @promatik , maybe I didn't express well myself. 

When I suggested cookies, was to avoid that ajax request, because cookies can be handled by both php and js ? Now there is the `httponly` property that some browsers support and allow to have cookies that cannot be manipulated with JS. 

Nowadays you ear google saying that is find a replacement for cookies ... so who knows.

I agree with you, the `flick` it's pretty much `unflickable`. I don't mind leaving this as is, it works good also in my opinion.

I would go with merge this, and if we decide it's worth moving to cookies instead of local storage do it for 4.2.

ping @tabacitu

**Comment by promatik** [comment]
@pxpm I haven't find anything related with `httponly` for javascript, but I found `document.cookie`, just did a small test and I was able to change the cookies by javascript 🙌
Didn't know about `document.cookie`, it's a great feature 😮

Problem with that; https://caniuse.com/?search=document.cookies
It's not widely supported, only chrome and edge  (60% worldwide).

So, should we implement the two solutions, and be ready for in a few month move to `document.cookie` only?
Or stick with just this one? And think about `document.cookie` in a few months?

Or maybe is there other option I haven't found?
Let me know ✌

**Comment by pxpm** [comment]
HttpOnly maybe it still in rfc proposal. https://owasp.org/www-community/HttpOnly

About `document.cookie` AFAIK mozilla also support it, but I migth be wrong. https://developer.mozilla.org/en-US/docs/Web/API/document/cookie

But I would have this one for now. I think we will probably never implement the cookies because it this ones works for our needs.

It's good that we are aware of the possibilities in case this bit us sometime in future :)

I will move this to almost ready, and let @tabacitu give the final call here. 

But for me it's a go here.


Best,
Pedro

File: src/resources/views/base/inc/sidebar.blade.php
Comment (tabacitu) [inline src/resources/views/base/inc/sidebar.blade.php:33]:
```suggestion
      // disable the transition animation temporarily, so that if you're browsing across
      // pages with the sidebar closed, the sidebar does not flicker into the view
```

File: src/resources/views/base/inc/sidebar.blade.php
Comment (tabacitu) [inline src/resources/views/base/inc/sidebar.blade.php:37]:
```suggestion
      // re-enable the transition, so that if the user clicks the hamburger menu, it does have a nice transition
```

File: src/resources/views/base/inc/sidebar.blade.php
Comment (tabacitu) [inline src/resources/views/base/inc/sidebar.blade.php:32]:
```suggestion
    if (sessionState) {
```

**Comment by tabacitu** [comment]
Haha excellent solution @promatik ! Great job, wouldn't have thought of that.

Totally agree with both you and @pxpm on this one:
- since this solution works on all browsers, let's merge it; we'll let _future us_ worry about maybe implementing it in a better way... someday... 😂
- an AJAX call sounds like overkill for this; not clean; _no bueno_; I'm happy you didn't go down that route (pun intended);

I've made some comment changes, and only have one question/worry. If you guys solve that please move it to "Ready to merge" and I will, ASAP.

Great collaboration here!

File: src/resources/views/base/inc/sidebar.blade.php
Comment (tabacitu) [inline src/resources/views/base/inc/sidebar.blade.php:28]: I see the query selector here is `.sidebar` - any chance people are using `.sidebar` somewhere else on page too, and we'd be eliminating their transition there?

I don't know, I'm just asking to be safe.

Comment (promatik) [reply src/resources/views/base/inc/sidebar.blade.php:28]: I did that "broad" query because on CSS, the transition is applied to that class with no other rules;
```css
.app-footer, .app-header, .aside-menu, .main, .sidebar {
    transition: margin-left .25s,margin-right .25s,width .25s,flex .25s;
}
```

But I agree, people may be using a sidebar on their dashboard for instance, we never know 🤷‍♂️ 

I'm going to change to `.app-body > .sidebar` since it's safer.

**Review by pxpm — APPROVED** [review]
100% agree changing to `app-body > sidebar` it will be more a more specific selector.

Like before, still working as expected.

It's not perfect, if you are with slow connection probably you still see the flick, like here https://recordit.co/IvyQi9hCr6 where the crud loaded is a bit heavy and DevTools with cache disabled so it had to reload the assets (not slow connection though)

I think in `general` this would not be a issue, and as said it's not 100% perfect, it would work pretty much 99.9% use cases.

**Comment by promatik** [comment]
@pxpm what if we implement both solutions? 😅

`document.cookie` is the correct one, but supported only on 60% of the browsers nowadays, this solution would be a fallback 🤷‍♂️
One day when all the browsers support `document.cookie` we would deprecate this method 🤷‍♂️

Let me know what you think.

**Comment by pxpm** [comment]
I don't think we should consider making both.

And as said before, and @tabacitu also agreed, this satisfy our needs for the moment. 100% we can `consider` doing it in 4.2. But for now I don't think we should spend more time here. 

If you think I am wrong please just let me know, we will handle it :)

**Comment by promatik** [comment]
I agree @pxpm 👌 let's move this for Ready to merge since you've already approved 🙌

**Pull request merged**
